From 79413c0b2fb33af4c29ac39b9ee423263a8eb61b Mon Sep 17 00:00:00 2001
From: skyel <razvan.ghitulete@gmail.com>
Date: Wed, 9 Apr 2014 18:06:34 +0200
Subject: [PATCH 14/18] edfi: disable LTO for conflicting arch

This patch disables LTO for some components in arch that either need
to be built as ELF (i.e. realmode and vdso) either have a great deal of
asm code that is discarded when compiled into bitcode

Signed-off-by: skyel <razvan.ghitulete@gmail.com>
---
 arch/x86/boot/Makefile        | 2 ++
 arch/x86/crypto/Makefile      | 7 +++++++
 arch/x86/kernel/acpi/Makefile | 3 +++
 arch/x86/lib/Makefile         | 2 ++
 arch/x86/power/Makefile       | 3 +++
 arch/x86/realmode/Makefile    | 3 +++
 arch/x86/realmode/rm/Makefile | 3 +++
 arch/x86/vdso/Makefile        | 4 ++--
 8 files changed, 25 insertions(+), 2 deletions(-)

diff --git a/arch/x86/boot/Makefile b/arch/x86/boot/Makefile
index fc3ccd6..24fda18 100644
--- a/arch/x86/boot/Makefile
+++ b/arch/x86/boot/Makefile
@@ -14,6 +14,8 @@
 # Set it to -DSVGA_MODE=NORMAL_VGA if you just want the EGA/VGA mode.
 # The number is the same as you would ordinarily press at bootup.
 
+LDFLAGS := $(PRISTINE_LDFLAGS)
+
 SVGA_MODE	:= -DSVGA_MODE=NORMAL_VGA
 
 targets		:= vmlinux.bin setup.bin setup.elf bzImage
diff --git a/arch/x86/crypto/Makefile b/arch/x86/crypto/Makefile
index 6c63c35..62dd829 100644
--- a/arch/x86/crypto/Makefile
+++ b/arch/x86/crypto/Makefile
@@ -2,8 +2,15 @@
 # Arch-specific CryptoAPI modules.
 #
 
+KBUILD_CFLAGS += $(DISABLE_LTO)
+LDFLAGS := $(PRISTINE_LDFLAGS)
+
 avx_supported := $(call as-instr,vpxor %xmm0$(comma)%xmm0$(comma)%xmm0,yes,no)
 
+
+
+
+
 obj-$(CONFIG_CRYPTO_ABLK_HELPER_X86) += ablk_helper.o
 obj-$(CONFIG_CRYPTO_GLUE_HELPER_X86) += glue_helper.o
 
diff --git a/arch/x86/kernel/acpi/Makefile b/arch/x86/kernel/acpi/Makefile
index 163b225..bc76394 100644
--- a/arch/x86/kernel/acpi/Makefile
+++ b/arch/x86/kernel/acpi/Makefile
@@ -1,3 +1,6 @@
+KBUILD_CFLAGS += $(DISABLE_LTO)
+LDFLAGS := $(PRISTINE_LDFLAGS)
+
 obj-$(CONFIG_ACPI)		+= boot.o
 obj-$(CONFIG_ACPI_SLEEP)	+= sleep.o wakeup_$(BITS).o
 
diff --git a/arch/x86/lib/Makefile b/arch/x86/lib/Makefile
index 96b2c66..b005755 100644
--- a/arch/x86/lib/Makefile
+++ b/arch/x86/lib/Makefile
@@ -2,6 +2,8 @@
 # Makefile for x86 specific library files.
 #
 
+KBUILD_CFLAGS += $(DISABLE_LTO)
+
 inat_tables_script = $(srctree)/arch/x86/tools/gen-insn-attr-x86.awk
 inat_tables_maps = $(srctree)/arch/x86/lib/x86-opcode-map.txt
 quiet_cmd_inat_tables = GEN     $@
diff --git a/arch/x86/power/Makefile b/arch/x86/power/Makefile
index a6a198c..8dd1ba5 100644
--- a/arch/x86/power/Makefile
+++ b/arch/x86/power/Makefile
@@ -1,5 +1,8 @@
 # __restore_processor_state() restores %gs after S3 resume and so should not
 # itself be stack-protected
+KBUILD_CFLAGS += $(DISABLE_LTO)
+LDFLAGS = $(PRISTINE_LDFLAGS)
+
 nostackp := $(call cc-option, -fno-stack-protector)
 CFLAGS_cpu.o	:= $(nostackp)
 
diff --git a/arch/x86/realmode/Makefile b/arch/x86/realmode/Makefile
index 94f7fbe..417569b 100644
--- a/arch/x86/realmode/Makefile
+++ b/arch/x86/realmode/Makefile
@@ -7,6 +7,9 @@
 #
 #
 
+LDFLAGS = $(PRISTINE_LDFLAGS)
+KBUILD_CFLAGS += $(DISABLE_LTO)
+
 subdir- := rm
 
 obj-y += init.o
diff --git a/arch/x86/realmode/rm/Makefile b/arch/x86/realmode/rm/Makefile
index cc5d4dd..5f7595e 100644
--- a/arch/x86/realmode/rm/Makefile
+++ b/arch/x86/realmode/rm/Makefile
@@ -9,6 +9,9 @@
 
 always := realmode.bin realmode.relocs
 
+KBUILD_CFLAGS += $(DISABLE_LTO)
+LDFLAGS := $(PRISTINE_LDFLAGS)
+
 wakeup-objs	:= wakeup_asm.o wakemain.o video-mode.o
 wakeup-objs	+= copy.o bioscall.o regs.o
 # The link order of the video-*.o modules can matter.  In particular,
diff --git a/arch/x86/vdso/Makefile b/arch/x86/vdso/Makefile
index 598f163..b96b1b9 100644
--- a/arch/x86/vdso/Makefile
+++ b/arch/x86/vdso/Makefile
@@ -3,6 +3,7 @@
 #
 
 KBUILD_CFLAGS += ${DISABLE_LTO}
+LDFLAGS = ${PRISTINE_LDFLAGS}
 
 VDSO64-$(CONFIG_X86_64)		:= y
 VDSOX32-$(CONFIG_X86_X32_ABI)	:= y
@@ -184,8 +185,7 @@ quiet_cmd_vdso = VDSO    $@
 		       -Wl,-T,$(filter %.lds,$^) $(filter %.o,$^) && \
 		 sh $(srctree)/$(src)/checkundef.sh '$(NM)' '$@'
 
-VDSO_LDFLAGS = -fPIC -shared $(call cc-ldoption, -Wl$(comma)--hash-style=sysv) \
-		${LTO_CFLAGS}
+VDSO_LDFLAGS = -fPIC -shared $(call cc-ldoption, -Wl$(comma)--hash-style=sysv) 
 GCOV_PROFILE := n
 
 #
-- 
1.8.3.1

